src/btree/bt_cursor.c

```diff
@@ -683,6 +683,14 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
         }
     }
 
+    /*
+     * If a valid key has been found and we are doing a prefix search near, we want to return the
+     * key only if it is a prefix match.
+     */
+    if (valid && F_ISSET(cursor, WT_CURSTD_PREFIX_SEARCH) &&
```

Comment:
@luke-pearson, do you think this is the right place for this check or should it be done in `__wt_cursor_valid`?

Reply:
I think its better to keep it here. The validity check is more concerned about record visibility than prefix key search semantics.